File: src/VL.FSharpDemoLib/03_Class.fs
namespace FSharpDemoLib
open System.Runtime.InteropServices //for the parameter attributes

type MyDataType (x:float32) =
    
    //private field
    let mutable FX = x
    let mutable FThreshold = 10.0f

    //private operations declared with 'let' will not show up
    let PrivateOp factor = 
        FX <- FX * factor
        FX

    //public mutable property
    member val Y = 0.0f with get, set

    //an operation called AddValue
    member this.AddValue (value) = 
        FX <- FX + value
        FX <- min FX FThreshold
        FX

    //another operation
    member this.SetThreshold ([<Optional; DefaultParameterValue(10)>] threshold) = 
        FThreshold <- threshold

File: VL.DemoLib_CSharp/03_Class.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DemoLib
{
    public class MyDataType
    {
        //private fields
        private float FX;
        private float FThreshold = 10f;

        //public property
        public float Y { get; set; }

        //constructor
        public MyDataType(float x)
        {
            FX = x;
        }

        //an operation called AddValue
        public float AddValue(float value)
        {
            FX += value;
            FX = Math.Min(FX, FThreshold);

            return FX;
        }

        //another operation
        public void SetThreshold(float threshold = 10f)
        {
            FThreshold = threshold;
        }

        //protected operations will not show up
        protected float ProtectedOp(float factor)
        {
            FX *= factor;
            return FX;
        }

        //private operations will not show up
        private float PrivateOp(float factor)
        {
            FX *= factor;
            return FX;
        }
    }
}
